docs/DeveloperGuide.md

```diff
@@ -1,38 +1,591 @@
+    - [Value proposition](#value-proposition)
+  - [User Stories](#user-stories)
+  - [Non-Functional Requirements](#non-functional-requirements)
+  - [Glossary](#glossary)
+  - [Instructions for manual testing](#instructions-for-manual-testing)
+  - [Instructions for JUnit Testing](#instructions-for-junit-testing)
+  - [Instructions for text-ui-testing](#instructions-for-text-ui-testing)
+  - [Future Enhancements](#future-enhancements)
+
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+LongAh uses the following libraries:
+
+1. [XChart](https://knowm.org/open-source/xchart/) - Used for generating charts to visualize data.
+
+LongAh uses the following tools for development:
+
+1. [JUnit 5](https://junit.org/junit5/) - Used for testing.
+2. [Gradle](https://gradle.org/) - Used for build automation.
+
+
+## Design & Implementation
+
+The UML diagram below provides an overview of the classes and their interactions within the LongAh application.
+
+![main.png](diagrams%2Fmain.png)
+
+Design and Implementation has been broken down into the subsequent sections, each tagged for ease of reference:
+
+* [UI and I/O](#ui-and-io)
+* [Commands](#commands)
+* [Storage](#storage)
+* [Member and MemberList](#member-and-memberlist)
+* [Transaction and TransactionList](#transaction-and-transactionlist)
+* [PIN](#pin)
+* [Chart](#chart)
+* [Exceptions and Logging](#exceptions-and-logging)
+
+### UI and I/O
+
+### Commands
+
+<ins>Overview</ins>
+
+The abstract `Command` class has been implemented to introduce an additional layer of abstraction between I/O and command execution, allowing for separation of handling command keywords and executing commands.
+
+<ins>Implementation Details</ins>
+
+![Command Inheritance Diagram](diagrams/CommandInheritance.png)
+
+<ins>Class Structure</ins>
+
+The abstract `Command` class and its related children classes have the following attributes:
+
+* *CommandString*: String indicating the command being parsed
+* *TaskExpression*: String containing details for the command to effect
+
+<ins>Constructor</ins>
+
+The Command constructor updates the attributes based on the input arguments.
+
+<ins>Methods</ins>
+
+The abstract `Command` class and its related children classes have the following method:
+* *execute*: Effect the command based on the `CommandString` and the `TaskExpression`
+
+### Storage
+
+<ins>Overview</ins>
+
+The `StorageHandler` class is responsible for managing the loading and saving of data regarding members and transactions from and onto the local machine. Each group calls its own `StorageHandler` object such that they maintain distinct storage directories.
+
+<ins>Implementation Details</ins>
+
+*Data Storage:*
+
+Each `StorageHandler` instance creates `members.txt` and `transactions.txt` in their respective folders. The file format is as follows.
```

Comment:
Suggest adding the path directory for where these text files are being saved in.